src/libmeasurement_kit/ooni/resources_impl.hpp

```diff
@@ -0,0 +1,227 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include <fstream>
+#include <regex>
+
+#include <measurement_kit/http.hpp>
+#include <measurement_kit/ooni.hpp>
+
+#include "../common/utils.hpp"
+
+namespace mk {
+namespace ooni {
+namespace resources {
+
+using namespace mk::http;
+
+static inline std::string get_base_url(const Settings &settings) {
+    return settings.get(
+        "ooni/resources_base_url",
+        std::string{
+            "https://github.com/OpenObservatory/ooni-resources/releases/"});
+}
+
+static inline void set_max_redirects(Settings &settings) {
+    if (settings.find("http/max_redirects") == settings.end()) {
+        // Note: 20 is what is used by web_connectivity and by browsers
+        settings["http/max_redirects"] = 20;
+    }
+}
+
+static inline std::string sanitize_version(const std::string &s) {
+    return std::regex_replace(s, std::regex{R"xx([\ \t\r\n]+)xx"}, "");
+}
+
+template <MK_MOCK_NAMESPACE(http, get)>
+void get_latest_release_impl(Callback<Error, std::string> callback,
+                             Settings settings, Var<Reactor> reactor,
+                             Var<Logger> logger) {
+    /*
+     * Note: this code does not use GitHub's "latest" release feature because
+     * upon creating a new release that is marked as latest but it does not
+     * contain yet all the files that are part of the release. For this reason
+     * we create a "latest" tag containing a "version" file that we only
+     * update when all the files are uploaded (#robustness). I'm adding this
+     * note here because I was initially confused by how this works.
+     *
+     *      - Simone (2016-12-06)
+     */
+    set_max_redirects(settings);
+    auto url = get_base_url(settings) + "/download/latest/version";
```

Comment:
You are actually adding an extra slash here in the URL path (it will become: `https://github.com/OpenObservatory/ooni-resources/releases//download/latest/version`)

I checked and the web server github is currently using doesn't care, but it's not a standard HTTP behaviour.